### ISSUE TITLE
Fix all issues that Debian or Ubuntu carry patches for

### ISSUE THREAD

**Issue by bp0** [lpereira/hardinfo#101]
https://anonscm.debian.org/git/collab-maint/hardinfo.git/tree/debian/patches
- [ ] make-build-reproducable.patch ?
- [x] fix-ppc64el-ftbfs.patch 
- [x] glibc-version.patch
- [ ] man page

**Comment by bp0** [comment]
@tsimonq2, there are better fixes here for ppc support and glib detection than are provided by those patches.

**Comment by tsimonq2** [comment]
@bp0 So you're saying that if I update to the latest Git snapshot in my packaging, I can drop `fix-ppc64el-ftbfs.patch` and `glibc-version.patch`?

As for the reproducibility patch, that's still a WIP and we're working with the Debian Reproducible Team to discuss the best route.

**Comment by bp0** [comment]
@tsimonq2, I am saying that. 
Although, pull/wait for #97 first. :smile: And maybe #98, #99...

**Comment by mckaygerhard** [comment]
either the  glibc-version.patch or the current detection on the git work today on debian, both process faild as in #79 see pictures .. recent commits of today make some difference?

**Comment by bp0** [comment]
The fixes were merged about a week ago, 10 July.
As I already mentioned, (https://github.com/lpereira/hardinfo/issues/79#issuecomment-314980276) your images show that you are not using a snapshot with the fixes applied.

**Comment by mckaygerhard** [comment]
ummm, i was using, but i'll retest and compile agains git right now! i compiled agains recent git! what i should do bad them?

![2017-07-20-105912_1366x768_scrot](https://user-images.githubusercontent.com/1571724/28424009-90c6712e-6d3a-11e7-8a73-30d45743872c.png)

**Comment by mckaygerhard** [comment]
the snapshot previous was agains a recent installation clean, the hardinfo was the firts compilation made in, its a recent machine

**Comment by bp0** [comment]
Anyway, the fix here is just an enhanced and more robust version of the Debian patch (use the first line of `ldconfig -V`, skip the first word). If it doesn't work with hardinfo master, it doesn't work with the Debian patch either. The Debian patch could still be dropped, and should be.

**Comment by mckaygerhard** [comment]
yeah the fix in git are a improved version of debian, so i tested that works in olders version of debian i tested in Debian lenny and debian squeeze,  all older versions of debian.. only fails on recent versions

**Comment by mckaygerhard** [comment]
https://anonscm.debian.org/git/collab-maint/hardinfo.git/tree/debian/patches

@tsimonq2:

- [ ]  make-build-reproducable.patch ?
- [x] fix-ppc64el-ftbfs.patch
- [x] glibc-version.patch : fixed now property: closed with 6209978 !
- [ ] man page : needs #54 be merged!

**Comment by bp0** [comment]
@tsimonq2 
Would #109 take care of `make-build-reproducable.patch`?

**Comment by tsimonq2** [comment]
@bp0 It's kind of hard to really determine that at this point (officially, because I don't have access to the infra in Debian) but it looks sane to me.

Otherwise, great work! Thanks for doing this. :)

**Comment by tsimonq2** [comment]
Also, I can take care of compiler-flags-hidden, from what I can see.

**Comment by bp0** [comment]
@tsimonq2 
It would be cool if an updated version could make it into Ubuntu 17.10.

**Comment by tsimonq2** [comment]
@bp0 As soon as I can drop all of the patches, I'll do an upload (that will automatically sync down to Ubuntu from Debian) with the latest Git snapshot.

To get it in Ubuntu, we have one month until Feature Freeze or there's extra paperwork for me (an exception to that freeze, which I will fill out if it comes to it). As soon as the man page PR is merged, I'll do that.

**Comment by mckaygerhard** [comment]
this issue are complety done! all task are done!

**Comment by bp0** [comment]
@tsimonq2 
The list is complete. Things are in a much better state here than the Debian package. It looks like ten days until Ubuntu's Debian import freeze, but doesn't it take five days for Debian to accept it into testing? Seems like it is getting close.
I'd like to see #147 included, but not super important.

**Comment by tsimonq2** [comment]
@bp0 ack :)

Also, Ubuntu autosyncs from Unstable, not Testing, so it should reach Ubuntu within a day of uploading to Debian.

**Comment by bp0** [comment]
Right on.

**Comment by tsimonq2** [comment]
An update to the hardinfo package based off of the latest Git snapshot has been uploaded to Debian Sid and Ubuntu Artful Aardvark. The package in Ubuntu Artful Aardvark has fixed the automated tests and is now installable.

As the Lubuntu Release Manager, I'll be sure to write something in my release notes for Lubuntu Artful Aardvark Beta 1 on August 31st. :wink: 

Please feel free to either file bugs against the Debian or Ubuntu hardinfo packages or ping me here if a bug has been fixed and you want me to include it in the package. Thanks!

**Comment by bp0** [comment]
Thanks! There already some bugs and fixes that should be included (ex: #156), but I guess this will always be true.

**Comment by bp0** [comment]
Here is a draft changelog: https://gist.github.com/bp0/5f35e76ad500a89c683f943fb1e1f235

**Comment by mckaygerhard** [comment]
hi bp0, you was so helfully for the project.. but i still waiting my coffe! uff

**Comment by bp0** [comment]
@tsimonq2 
There was one rather obvious bug (#156) fixed since your last upload, and a few improvements. Could you upload a new package?

**Comment by tsimonq2** [comment]
@bp0 Submitted to my sponsors.

**Comment by tsimonq2** [comment]
@bp0 My sponsors (rightfully) pointed out that uber-graph being embedded in the source breaks the DFSG (Debian Free Software Guidelines) and that I'll need to package uber-graph as a separate source. I'll do that, and in the meantime (the Debian NEW queue takes a month or two), could hardinfo upstream please adapt to either not include the uber-graph source by default or to at least make it adaptable so if I exclude that part of the source in the hardinfo package, it can look for the uber-graph libraries?

**Comment by bp0** [comment]
Ubergraph is only used by the gtk3 version. Hardinfo uses gtk2 by default.
If ubergraph is left out of the source, hardinfo will still build without
any problems in the default configuration.

I'll look into the rest when I get a chance.

On Sep 16, 2017 1:01 PM, "Simon Quigley" <notifications@github.com> wrote:

@bp0 <https://github.com/bp0> My sponsors (rightfully) pointed out that
uber-graph being embedded in the source breaks the DFSG (Debian Free
Software Guidelines) and that I'll need to package uber-graph as a separate
source. I'll do that, and in the meantime (the Debian NEW queue takes a
month or two), could hardinfo upstream please adapt to either not include
the uber-graph source by default or to at least make it adaptable so if I
exclude that part of the source in the hardinfo package, it can look for
the uber-graph libraries?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/lpereira/hardinfo/issues/101#issuecomment-329985085>,
or mute the thread

**Comment by mckaygerhard** [comment]
for this isue clodes only the uber-grap issue ? if there only all the "carry patches" all are done some time ago ! in that case this issue can be close!
